flink-libraries/flink-table/src/main/scala/org/apache/flink/table/expressions/time.scala

```diff
@@ -328,6 +330,42 @@ case class TemporalOverlaps(
   }
 }
 
+  /**
+    * Standard conversion of the TIMESTAMPADD operator.
+    * Source: [[org.apache.calcite.sql2rel.StandardConvertletTable#TimestampAddConvertlet]]
+    */
+case class TimestampAdd(
+    unit: Expression,
+    count: Expression,
+    timestamp: Expression)
+  extends Expression {
+
+  override private[flink] def children = unit :: count :: timestamp :: Nil
+
+  override private[flink] def toRexNode(implicit relBuilder: RelBuilder) = {
+    var timeUnit : Option[TimeUnit] = None
+    if (unit.isInstanceOf[Literal]) {
```

Comment:
I think if we use pattern matching here it will be much cleaner, something like.
```
    unit match {
      case literal: Literal => //...
      case _ => //... 
   }
```
That also remind me we should probably override `validateInput` function as well.